Compiler/test/verifytrim.jl

```diff
@@ -18,31 +18,42 @@ let infos = Any[]
     @test isempty(parents)
 end
 
+make_cfunction() = @cfunction(+, Float64, (Int64,Int64))
+
 # use TRIM_UNSAFE to bypass verifier inside typeinf_ext_toplevel
-let infos = typeinf_ext_toplevel(Any[Core.svec(Base.SecretBuffer, Tuple{Type{Base.SecretBuffer}})], [Base.get_world_counter()], TRIM_UNSAFE)
-    @test_broken length(infos) > 4
+let infos = typeinf_ext_toplevel(Any[Core.svec(Ptr{Cvoid}, Tuple{typeof(make_cfunction)})], [Base.get_world_counter()], TRIM_UNSAFE)
     errors, parents = get_verify_typeinf_trim(infos)
     @test_broken isempty(errors) # missing cfunction
-    #resize!(infos, 4)
-    #errors, = get_verify_typeinf_trim(infos)
 
     desc = only(errors)
     @test desc.first
     desc = desc.second
     @test desc isa CallMissing
-    @test occursin("finalizer", desc.desc)
+    @test occursin("cfunction", desc.desc)
     repr = sprint(verify_print_error, desc, parents)
     @test occursin(
-        r"""^unresolved finalizer registered from statement \(Core.finalizer\)\(Base.final_shred!, %new\(\)::Base.SecretBuffer\)::Nothing
```

Comment:
We should put back this test, for code coverage sake, both for SecretBuffer (working) and for something else (brokeN). If necessary (since this is presumably fixed), we can change the initial inferred call to something like `Core.svec(finalizer, Tuple{identity, Any})`

Reply:
The working test for SecretBuffer is on https://github.com/JuliaLang/julia/pull/58014/files/0578395d9c980f496b188a383cc9266bce86d689#diff-e3e801c6894fd826c1162f7037c483ed0df8955c10a29b25cbaa22d538482e7bR53

but adding coverage for a failed `finalizer` verification sounds like an excellent idea